tests/integration/test_interface.py

```diff
@@ -78,53 +111,71 @@ def test_generate_dataset_from_sample_and_source(
-        # ACS and CPS datasets (very small)
-        if len(compare_sample_idx) < 30 or len(compare_dataset_idx) < 30:
-            warnings.warn(
-                f"Noise levels in {col} of {dataset_name} were not compared because the numbers of rows eligible "
-                f"for noise were only {len(compare_sample_idx)} and {len(compare_dataset_idx)}"
+
+    config = get_configuration(config)
+    for col_name in check_noised_sample.columns:
+        col = COLUMNS.get_column(col_name)
+
+        # Check that originally missing data remained missing
+        originally_missing_sample_idx = check_original_sample.index[
+            check_original_sample[col.name].isna()
+        ]
+        originally_missing_dataset_idx = check_original_dataset.index[
+            check_original_dataset[col.name].isna()
+        ]
+        assert check_noised_sample.loc[originally_missing_sample_idx, col.name].isna().all()
+        assert check_noised_dataset.loc[originally_missing_dataset_idx, col.name].isna().all()
+
+        # Check for noising where applicable
+        to_compare_sample_idx = shared_sample_idx.difference(originally_missing_sample_idx)
+        to_compare_dataset_idx = shared_dataset_idx.difference(originally_missing_dataset_idx)
+        if col.noise_types:
+            # Note: Coercing check_original to string. This seems like it should not
+            # have passed before but our rtol was 0.7
+            if col.name in INT_COLUMNS:
+                check_original_sample[col.name] = cleanse_integer_columns(
+                    check_original_sample[col.name]
+                )
+                check_original_dataset[col.name] = cleanse_integer_columns(
+                    check_original_dataset[col.name]
+                )
+
+            noise_level_sample = (
+                check_original_sample.loc[to_compare_sample_idx, col.name].values
+                != check_noised_sample.loc[to_compare_sample_idx, col.name].values
+            ).sum()
+            noise_level_dataset = (
+                check_original_dataset.loc[to_compare_dataset_idx, col.name].values
+                != check_noised_dataset.loc[to_compare_dataset_idx, col.name].values
+            ).sum()
+
+            # Validate column noise level
+            _validate_column_noise_level(
+                dataset_name=dataset_name,
+                check_data=check_original_sample,
+                check_idx=to_compare_sample_idx,
+                noise_level=noise_level_sample,
+                col=col,
+                config=config,
+                fuzzy_name="test_generate_dataset_from_sample_and_source_sample",
+                validator=fuzzy_checker,
+            )
+            _validate_column_noise_level(
+                dataset_name=dataset_name,
+                check_data=check_original_dataset,
+                check_idx=to_compare_dataset_idx,
+                noise_level=noise_level_dataset,
+                col=col,
+                config=config,
```

Comment:
From discussion, I do think that we still need a test to check the sample noising vs the dataset noising. Maybe that's back to just a simple `np.isclose(...)`?

Reply:
Curious about this. I don't see what it adds to compare them directly, but I agree that comparing them directly doesn't fit into the FuzzyChecker assert method that is already implemented.

Reply:
We want to compare them directly to make sure when we noise multiple shards vs one dataset the same amount of noise is happening. Maybe the name of the test should be changed to be more clear or something I am not sure. This test is doing 2 things:
1. We are testing if we have a large dataframe and then split it into multiple shards and then noise both of those we can validate we are experiencing the same amount of noise.
2. It indirectly is testing that noising multiple shards works. 
I think this implementation adequately tests both.

Reply:
But if we just check that each one is noising the expected amount, we are testing that noise with one shard = expected amount = noise with multiple shards. Is that not _stricter_ than testing noise with one shard = noise with multiple shards?

Reply:
Yes but doing the fuzzy checker on both a single dataset and multiple shards doesn't directly compare them we are just making that assumption.

Reply:
Yes, it doesn't directly compare them, but I don't see how directly comparing them adds anything.

With how the test used to be, we could have a horrible bug where both versions (single shard and multi-shard) got noised in 100% of the rows, and this test would pass. Now, it checks that the single shard is close enough to the expected noise, and that the multi-shard is close enough to the expected noise. The test wouldn't pass if they were both 100%, since the expected noise will be much lower than that.

Another way to say it: we expect the randomness to be completely independent between the single shard and multi-shard case. The only _reason_ they should be close is because they are both close to the expected noise. Why not just test that directly?

Reply:
I've added the other validation back in some we are testing both cases now.

Reply:
If we're comparing both versions to the expected value, there's not reason to compare them to each other.